x-pack/plugins/cross_cluster_replication/public/app/services/auto_follow_pattern_validators.js

```diff
@@ -88,7 +88,25 @@ export const validateLeaderIndexPattern = (indexPattern) => {
     };
   }
 
-  return { message: null };
+  return null;
+};
+
+export const validateLeaderIndexPatterns = (indexPatterns) => {
+  if (!indexPatterns.length) {
+    return {
+      message: i18n.translate('xpack.crossClusterReplication.autoFollowPattern.leaderIndexPatternValidation.isEmpty', {
+        defaultMessage: 'At least one leader index pattern is required.',
+      })
+    };
+  }
+
+  let error = null;
+  indexPatterns.forEach((indexPattern) => {
+    if (!error) {
```

Comment:
The `forEach` will keep looping unnecessarily, even after an error has been found. If we just want to find the first error, how about using a `for` loop?

```js
for (let i = 0; i < indexPatterns.length; i++) {
  const error = validateLeaderIndexPattern(indexPatterns[i]);
  if (error) {
    return error;
  }
}

return null;
```